course/models.py

```diff
@@ -964,11 +964,13 @@ def enroll_from_sis(self) -> Tuple[int, int]:
             # unenrolled themselves.
             students = self.all_students.filter(enrollment__from_sis=True)
             to_remove = students.exclude(student_id__in=participants)
-            qs = Enrollment.objects.filter(user_profile__in=to_remove, course_instance=self)
-            qs.update(status=Enrollment.ENROLLMENT_STATUS.REMOVED)
+            qs = (Enrollment.objects
+                  .filter(user_profile__in=to_remove, course_instance=self)
+                  .exclude(status=Enrollment.ENROLLMENT_STATUS.REMOVED)
+            )
             for e in qs:
                 invalidate_content(Enrollment, e)
-                delcount += 1
+            delcount = qs.update(status=Enrollment.ENROLLMENT_STATUS.REMOVED)
```

Comment:
Would it still make more sense to remove the students before invalidating the cache? `qs.update()` used to come before `invalidate_content()`.

Reply:
I tried that first, but it seems that after the update operation the qs becomes empty here. Apparently the update can affect the qs contents if the same attribute that is updated is also used for filtering.

Reply:
Right, I didn't think of that. At least, one possibility is to copy the original list before running `update()`. 
```
qs_before_update = list(qs)
delcount = qs.update(status=...)
for e in qs_before_update:
    invalidate_content(...)
```

Do you prefer to change this or merge the pull request without further changes?

Reply:
I don't have strong preference, so either option is fine with me, but is the order between invalidate and update that important? Compared to having a new variable for a temporary list?

Reply:
It is not a big deal. We can keep it like you originally did.